extensions/wikibase/src/org/openrefine/wikibase/schema/WbDateVariable.java

```diff
@@ -57,13 +59,18 @@ public TimeValue fromCell(Cell cell, ExpressionContext ctxt)
         if (cell == null || cell.value == null) {
             throw new SkipSchemaExpressionException();
         }
+        String value = cell.value.toString();
         try {
             // parsed dates are accepted by converting them to strings
-            return WbDateConstant.parse(cell.value.toString());
+            return WbDateConstant.parse(value);
         } catch (ParseException e) {
-            if (!cell.value.toString().isEmpty()) {
+            if (value.equals("#NOVALUE#")) {
+                throw new SpecialValueNoValueException();
+            } else if (value.equals("#SOMEVALUE#")) {
+                throw new SpecialValueSomeValueException();
+            } else if (!value.isEmpty()) {
```

Comment:
Do we actually need special handling here given that you have added support in `WbVariableExpr` already?

Reply:
@wetneb I'm afraid so as parsing appears to be used for things like QA. I also think such things needs to be added to many more data types :-/

Reply:
I do not really understand why - could it be that your implementation in `WbVariableExpr` did not work so far because of the `==` instead of the `.equals()`? And what does this have to do with QA?

Reply:
> I do not really understand why - could it be that your implementation in WbVariableExpr did not work so far because of the == instead of the .equals()? 

That might indeed be it.

> And what does this have to do with QA?

Given a magic keyword the `!value.isEmpty()` will raise a QA Warning/Error.

Reply:
> >I do not really understand why - could it be that your implementation in WbVariableExpr did not work so far because of the == instead of the .equals()?

>That might indeed be it.

That does indeed take care of the issue!